### PR TITLE
Python raster reader argument refactor

### PR DESCRIPTION
**Pull request by vpipkt**
Initial argument for `spark.read.raster` is called  `source` and can now be a str, list of str, list of list of str, pandas DF catalog or spark DF catalog.

Add docstring to this important entry point for users.

Simplifies documentation pages to remove unneeded `catalog` named argument.

Expands unit test in python to all claimed use cases of `source`.

### CODE REVIEW

File: pyrasterframes/src/main/python/pyrasterframes/__init__.py [hunk 1]
```diff
@@ -110,16 +110,33 @@ def _aliased_writer(df_writer, format_key, path, **options):
...
-        path=None,
-        catalog=None,
+        source=None,
         catalog_col_names=None,
         band_indexes=None,
         tile_dimensions=(256, 256),
         lazy_tiles=True,
         **options):
+    """
+    Returns a Spark DataFrame from raster data files specified by URI pointers
```
Comment (metasim) [inline pyrasterframes/src/main/python/pyrasterframes/__init__.py]: End with `.`?

Also, "pointers" may be loaded... can it just say "by URIs"?

File: pyrasterframes/src/main/python/pyrasterframes/__init__.py [hunk 2]
```diff
@@ -110,16 +110,33 @@ def _aliased_writer(df_writer, format_key, path, **options):
...
+    The returned DataFrame will have a column of (CRS, Extent, Tile) for each URI read
+    Multiple bands from the same raster file are spread across rows of the DataFrame. See band_indexes param.
```
Comment (metasim) [inline pyrasterframes/src/main/python/pyrasterframes/__init__.py]: Pedantic: `band_indexes`.

File: pyrasterframes/src/main/python/pyrasterframes/__init__.py [hunk 3]
```diff
@@ -110,16 +110,33 @@ def _aliased_writer(df_writer, format_key, path, **options):
...
+    If bands from a scene are stored in separate files, provide a DataFrame to the `source` parameter. Each row in the returned DataFrame will contain one (CRS, Extent, Tile) for each item in `catalog_col_names`
+
+    For more details and example usage, consult https://rasterframes.io/raster-read.html
+
+    :param source: a string, list of strings, list of lists of strings, a pandas DataFrame or a Spark DataFrame giving URIs to the raster data to read
```
Comment (metasim) [inline pyrasterframes/src/main/python/pyrasterframes/__init__.py]: Pendantic: _pandas_ -> _Pandas_

File: pyrasterframes/src/main/python/pyrasterframes/__init__.py (hunk 1 above)
Comment (metasim) [reply pyrasterframes/src/main/python/pyrasterframes/__init__.py]: Note: I don't really know the convention with sentences/periods in Python docs. My inclination would be to provide standard grammatical termination.

File: pyrasterframes/src/main/python/tests/RasterFunctionsTests.py [hunk 4]
```diff
@@ -286,7 +286,7 @@ def test_render_composite(self):
         cat = self.spark.createDataFrame([
             Row(red=self.l8band_uri(4), green=self.l8band_uri(3), blue=self.l8band_uri(2))
         ])
-        rf = self.spark.read.raster(catalog = cat, catalog_col_names=['red', 'green', 'blue'])
+        rf = self.spark.read.raster(catalog=cat, catalog_col_names=cat.columns)
```
Comment (metasim) [inline pyrasterframes/src/main/python/tests/RasterFunctionsTests.py]: In other examples you've dropped the named keyword parameter.... are you testing that `catalog` still works here, or should it be dropped?

File: pyrasterframes/src/main/python/tests/RasterSourceTest.py [hunk 5]
```diff
@@ -18,21 +18,38 @@
...
 
-
 from pyrasterframes.rasterfunctions import *
+from pyrasterframes.rf_types import *
+from pyspark.sql.functions import *
+import os.path
+from unittest import skip
 from . import TestEnvironment
 
-class RasterSource(TestEnvironment):
```
Comment (metasim) [inline pyrasterframes/src/main/python/tests/RasterSourceTest.py]: Heh... funny that slipped through so long.

File: pyrasterframes/src/main/python/tests/RasterSourceTest.py [hunk 6]
```diff
@@ -18,21 +18,38 @@
...
+        }
+
+        assert band in range(1, 12)
+        assert scene in scene_dict.keys()
+        p = scene_dict[scene]
+        return p.format(band)
 
     def test_handle_lazy_eval(self):
-        df = self.spark.read.raster(self.img_uri)
+        df = self.spark.read.raster(self.path(1, 1))
```
Comment (metasim) [inline pyrasterframes/src/main/python/tests/RasterSourceTest.py]: 🙏 Thank you! I'd like to eventually purge that `img_uri` member.

File: pyrasterframes/src/main/python/tests/RasterSourceTest.py [hunk 7]
```diff
@@ -18,21 +18,38 @@
...
+import os.path
+from unittest import skip
 from . import TestEnvironment
 
-class RasterSource(TestEnvironment):
+
+class RasterSourceTest(TestEnvironment):
+
+    @staticmethod
+    def path(scene, band):
```
Comment (metasim) [inline pyrasterframes/src/main/python/tests/RasterSourceTest.py]: Wondering if this should be promoted to `TestEnvironment`...

File: pyrasterframes/src/main/python/tests/RasterFunctionsTests.py (hunk 4 above)
Comment (vpipkt) [reply pyrasterframes/src/main/python/tests/RasterFunctionsTests.py]: yes for back compatibility it is good to have at least one ; but probably better to do it in a single unit test so the intent is clear.

**Comment by vpipkt** [comment]
@metasim I made some changes to data source parameters, which hopefully simplifies things in the direction you were thinking.

I am less clear on the vision for synchronizing naming between scala and python. Happy to discuss

**Pull request merged**